test/unit/org/apache/cassandra/service/snapshot/TableSnapshotTest.java

```diff
@@ -298,7 +298,7 @@ public void testShouldClearSnapshot() throws Exception
                 // 1. snapshot to clear is not ephemeral
                 // 2. tag to clear is null, empty, or it is equal to snapshot tag
                 // 3. byTimestamp is true
-                if (TableSnapshot.shouldClearSnapshot(testingTag, olderThanTimestamp).test(snapshot))
+                if (SnapshotManager.shouldClearSnapshot(testingTag, Set.of(keyspace), olderThanTimestamp, false).test(snapshot))
```

Comment:
this is another behavior change in the test and I'm not sure if it's desired. Should we avoid including the keyspace? The comments above seem to indicate we don't want to include the keyspace here
```suggestion
                if (SnapshotManager.shouldClearSnapshot(testingTag, Set.of(), olderThanTimestamp, false).test(snapshot))
```

Reply:
I think this is just fine. `shouldClearSnapshot` method indeed accepts keyspace now, but the logic in it takes into account the situation when a keyspaces set is empty, preserving the original behaviour.